### PR TITLE
feat(nuxt): full scoped slots support for server components

### LINKED ISSUE #19765: support client interactivity within server components
### Describe the feature

Right now, you can't have client components as children of server components, but enabling this would enable a bunch of usful usecases.

for example I have a big website using nuxt, with mostly static content but also bit of interactive content (like accordions or toggles to hide and show content).
Server components would allow me to not send the JS for most of my site to the client, greatly reducing time spend on script parsing and execution (hydration), but right now that would disallow me from using any interactive elements.
(for reference, next.js supports this: https://beta.nextjs.org/docs/rendering/server-and-client-components#moving-client-components-to-the-leaves)

I would imagine that the easiest way would be to enable server components to have slots and in the client somehow check if there is a client component slot present in the server component html.
Maybe NuxtServerComponent needs to parse the html to find out where to inject the slots or we can store that info in the island response.

I have a bunch of work time next week to dedicate to working on this, but I dont really know where to start, if anybody has any pointers where I can look or what the best approach would be, that would be appreciated, I am MiniDigger#3086 on discord.

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### LINKED ISSUE #19884: Islands components not updating in production
### Environment

All 

from Nuxt 3.3.0 to current edge version

### Reproduction

https://stackblitz.com/edit/github-qwz87m?file=package.json,components%2Fislands%2FTest.vue,app.vue,nuxt.config.ts

npm run build 
node .output/server/index.mjs

Click on  `add + 1`

the count is not updated

### Describe the bug

It seems that we have some issues with updating the render of a `NuxtIsland` and `.server` (only) components in production.

It may comes from the usage of `createStaticVNode` ?

The render function is correctly triggered from `NuxtIsland` but DOM nodes are not getting updated

https://github.com/nuxt/nuxt/pull/19605 did not solve this issue

### Additional context
At `setupRenderEffect`, the `nextTree` doesn't have any dom node at `el` in production. In dev mode, the `el` represents a `Fragment` marker `<!--[-->`

### Logs

_No response_

### PR DESCRIPTION
**Pull request by huang-julien**
### 🔗 Linked issue
resolve https://github.com/nuxt/nuxt/issues/19765
resolve https://github.com/nuxt/nuxt/issues/19884
#19772 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

hey there :wave: !

This PR is forked from my previous PR https://github.com/nuxt/nuxt/pull/19851 since it is ready for basic slots. 

This changes some things from #19851 to transform island components slots at build time instead and allow to have:
- conditionnal slots with v-if
- scoped slots
- slots with v-for

Let me know if you think that a case is missing about server-only components slots. 

Of course, https://github.com/nuxt/nuxt/pull/19851 will be closed if this PR is ready before it's merged

### How does it work

please check https://github.com/nuxt/nuxt/pull/19851 for previous implementation.

- kept the same idea of using `Teleport` in `NuxtIsland`
- transform all `<slot>` at build time, replacing it by a container div.
- transform all bindings and `v-for` directive to generate an array in its attribute so the slots within `NuxtIsland` can map over it to render
- we now use `nuxt-*` attributes for ids slots names and scoped data

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly (todo).

### CODE REVIEW

File: packages/nuxt/src/app/components/nuxt-island.ts [hunk 1]
```diff
@@ -28,15 +32,37 @@ export default defineComponent({
...
+    const ssrHTML = ref<string>(process.client ? getFragmentHTML(instance.vnode?.el ?? null).join('') ?? '<div></div>' : '<div></div>')
+    const uid = ref<string>(ssrHTML.value.match(SSR_UID_RE)?.[1] ?? randomUUID())
+    const availableSlots = computed(() => {
+      return [...ssrHTML.value.matchAll(SLOTNAME_RE)].map(m => m[1])
+    })
 
-    const html = ref<string>(process.client ? getFragmentHTML(instance?.vnode?.el).join('') ?? '<div></div>' : '<div></div>')
+    const html = computed(() => {
+      const currentSlots = Object.keys(slots)
+      return ssrHTML.value.replaceAll(SLOT_FALLBACK_RE, (full, slotName, content) => {
```
Comment (danielroe) [inline packages/nuxt/src/app/components/nuxt-island.ts:50]:
```suggestion
      return ssrHTML.value.replace(SLOT_FALLBACK_RE, (full, slotName, content) => {
```

File: packages/nuxt/src/components/runtime/server-component.ts [hunk 2]
```diff
@@ -96,11 +107,57 @@ const NuxtServerComponent = defineComponent({
...
+    })
+    const availableSlots = computed(() => {
+      return [...res.data.value!.html.matchAll(SLOTNAME_RE)].map(m => m[1])
+    })
+
+    const html = computed(() => {
+      const currentSlots = Object.keys(slots)
+      return res.data.value!.html.replace(UID_ATTR, () => {
+        return `nuxt-ssr-component-uid="${randomUUID()}"`
+      }).replaceAll(SLOT_FALLBACK_RE, (full, slotName, content) => {
```
Comment (danielroe) [inline packages/nuxt/src/components/runtime/server-component.ts:129]:
```suggestion
      }).replace(SLOT_FALLBACK_RE, (full, slotName, content) => {
```

Comment (danielroe) [inline packages/nuxt/src/components/runtime/server-component.ts:129]:
```suggestion
      return res.data.value!.html
        .replace(UID_ATTR, () => `nuxt-ssr-component-uid="${randomUUID()}"`)
        .replace(SLOT_FALLBACK_RE, (full, slotName, content) => {
```

**Review by danielroe — APPROVED** [review]

**Comment by danielroe** [comment]
This is great work. Let's iterate on it once merged, as I suspect there is some scope here for reduction in complexity, particularly once we move server components out of the payload.

**Comment by rczdgbg** [comment]
When will v3.6.0 be released?

**Comment by danielroe** [comment]
If you are looking for this PR, it is released in v3.5.0. But v3.6.0 should be released on or around June 20: https://github.com/nuxt/nuxt/milestone/5.

**Comment by bernhardberger** [comment]
This doesn't allow us to have interactive client components in slots though, correct? Anything inside an island component is still rendered server side only without hydration?

Usually it's the wrapper components (components for Layouts and stuff) that have no need to be hydrated on the client side and cause JS size to explode. Is that something that will be possible one day? That would be huge.

**Comment by danielroe** [comment]
It is indeed. An experimental version of it will be in v3.9.

**Pull request merged**